client/src/views/BlockDetail.vue

```diff
@@ -142,48 +153,63 @@ export default {
       blockInfo: {},
       tableData: {
         head: [
-          '#',
-          'Timestamp',
-          'Transaction Type',
-          'Fees',
-          'Signer',
-          'Transaction Hash'
+          "#",
+          "Timestamp",
+          "Transaction Type",
+          "Fees",
+          "Signer",
+          "Transaction Hash"
         ],
         data: []
       },
       loading: true
-    }
+    };
+  },
+  watch: {
+    $route: "asyncData"
   },
   mounted() {
-    this.asyncData()
+    this.asyncData();
   },
   methods: {
+    previousBlcok() {
+      var previousBlcok = Number(this.blockId) - 1;
+      this.$router.push({
+        path: "/block/" + previousBlcok
+      });
+    },
+    nextBlock() {
+      var nextBlock = Number(this.blockId) + 1;
+      this.$router.push({
+        path: "/block/" + nextBlock
+      });
+    },
     asyncData() {
-      let self = this
-      this.blockId = this.$route.params.blockId
-      this.loading = 0
-      DataService.getBlockInfo(this.blockId).then(function (data) {
-        self.blockInfo = data.blockInfo
+      let self = this;
+      this.blockId = this.$route.params.blockId;
+      this.loading = 0;
+      DataService.getBlockInfo(this.blockId).then(function(data) {
+        self.blockInfo = data.blockInfo;
         if (data.blockTransactionList.length) {
-          self.tableData.data = []
+          self.tableData.data = [];
           data.blockTransactionList.forEach((el, idx) => {
-            let temp = []
-            let singerLink = `<a href="/#/account/${el.signer}">${el.signer}</a>`
-            let transactionHashLink = `<a href="/#/transaction/${el.transactionHash}">${el.transactionHash}</a>`
-            temp.push(idx + 1)
-            temp.push(el.deadline)
-            temp.push(el.transactionBody.type)
-            temp.push(el.fee)
-            temp.push(singerLink)
-            temp.push(transactionHashLink)
-            self.tableData.data.push(temp)
-          })
+            let temp = [];
+            let singerLink = `<a href="/#/account/${el.signer}">${el.signer}</a>`;
+            let transactionHashLink = `<a href="/#/transaction/${el.transactionHash}">${el.transactionHash}</a>`;
```

Comment:
This line does not look right, the js code should not be creating hardcoded DOM elements. One option is to pass the necessary parameters to ``self.tableData`` and the HTML reconstructs the URL.

Reply:
will find a solution to let  the template construct the URL